pkg/nuclide/ui/tree/lib/TreeRootComponent.js

```diff
@@ -178,6 +190,35 @@ var TreeRootComponent = React.createClass({
     this.setState({expandedKeys});
   },
 
+  async _toggleChildNodesExpanded(parentNode: LazyTreeNode, forceExpanded: ?boolean): Promise {
```

Comment:
Check out `expandNodeKey` and `collapseNodeKey`; I think the functionality there can be reused to reduce the amount of new code that needs to be rewritten here. Expanding all children might be better as options to those methods to limit new methods on this class too.